core/syncronizer/src/syncronizer.rs

```diff
@@ -58,8 +58,10 @@ impl<C: NodeComponents> Syncronizer<C> {
 
         // We only run the prelude in prod mode to avoid interfering with tests, and if genesis
         // has already been applied, otherwise we skip it.
-        if !cfg!(debug_assertions) && query_runner.has_genesis() {
```

Comment:
Isn't the genesis applied at the beginning of the e2e tests as well? https://github.com/fleek-network/lightning/blob/54ff8da10b47959d6b7493fc4ae1e653213b074d/core/e2e/src/swarm.rs#L400

Reply:
Yep this genesis check is for the [test-utils/e2e](https://github.com/fleek-network/lightning/blob/main/core/test-utils/src/e2e/network_builder.rs) utils where the node supports starting up without a genesis, discovers the random listening ports of various components, builds the genesis from that and applies it, while components that need it are just waiting for genesis to be applied before moving forward after start. So that let's us do things like create multi node e2e-like tests with random ports.

Reply:
Okay cool. But now we aren't skipping the synchronizer prelude for e2e tests (not test-utils/e2e), right?

Reply:
Yep correct, but what seems to be happening sometimes in the tests that have non-committee nodes is that the bootstrap nodes aren't available yet when syncronizer init/prelude is called, so this commit makes it so that the rest of the prelude is skipped when that happens.

I'm just realizing now that this only happens in the [e2e_epoch_change_with_some_nodes_not_on_committee](https://github.com/snormore/lightning/blob/s/e2e-tests-cleanup/core/e2e/tests/epoch_change.rs#L57C10-L57C59) test, and not the [e2e_syncronize_state](https://github.com/snormore/lightning/blob/s/e2e-tests-cleanup/core/e2e/tests/syncronizer.rs) test, because in the syncronizer test we launch all the committee nodes first, then the non-committee node after they're up, so we avoid that issue, which makes sense.

Reply:
Maybe we actually want non-committee nodes to error out during `init` if no bootstrap nodes were available... 

I'm going to reverse out this commit and just update the test to startup the non-committee node after the others.

Reply:
Yeah. I added this debug check to skip the entire prelude for e2e tests to avoid this race. But seems like we have this case covered now. Thanks for the clarifications :+1:

Reply:
Realizing I missed your other question:
> Are we running the e2e tests in release mode now?

We run all the tests in release mode in CI/`flake.nix`, so I didn't even realize it was happening until after I opened the PR because I was running them in debug mode locally